backup-s3/src/main/scala/io/aiven/guardian/kafka/backup/s3/BackupClient.scala

```diff
@@ -126,29 +127,39 @@ class BackupClient[T <: KafkaClientInterface](maybeS3Settings: Option[S3Settings
   private[s3] def successSink
       : Sink[(SuccessfulUploadPart, immutable.Iterable[kafkaClientInterface.CursorContext]), Future[Done]] =
     kafkaClientInterface.commitCursor
-      .contramap[(SuccessfulUploadPart, immutable.Iterable[kafkaClientInterface.CursorContext])] { case (_, cursors) =>
-        kafkaClientInterface.batchCursorContext(cursors)
+      .contramap[(SuccessfulUploadPart, immutable.Iterable[kafkaClientInterface.CursorContext])] {
+        case (part, cursors) =>
+          logger.info(
+            s"Committing kafka cursor for uploadId:${part.multipartUpload.uploadId} key: ${part.multipartUpload.key} and S3 part: ${part.partNumber}"
+          )
+          kafkaClientInterface.batchCursorContext(cursors)
       }
 
   private[s3] def kafkaBatchSink
-      : Sink[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext]), NotUsed] = {
-
-    val success = Flow[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext])]
-      .collectType[(SuccessfulUploadPart, immutable.Iterable[kafkaClientInterface.CursorContext])]
-      .wireTap { data =>
-        val (part, _) = data
-        logger.info(
-          s"Committing kafka cursor for uploadId:${part.multipartUpload.uploadId} key: ${part.multipartUpload.key} and S3 part: ${part.partNumber}"
+      : Sink[(UploadPartResponse, immutable.Iterable[kafkaClientInterface.CursorContext]), NotUsed] =
+    // Taken from https://doc.akka.io/docs/akka/current/stream/operators/Partition.html
```

Comment:
Maybe add the license and the authors on the top of the file. Correct attributions is good